pupil_src/shared_modules/calibration_routines/data_processing.py

```diff
@@ -0,0 +1,203 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2020 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+def filter_confidence(pupil_list, threshold):
+    len_pre_filter = len(pupil_list)
+
+    pupil_list = [p for p in pupil_list if p["confidence"] >= threshold]
+
+    len_post_filter = len(pupil_list)
+    try:
+        dismissed_percentage = 100 * (1.0 - len_post_filter / len_pre_filter)
+    except ZeroDivisionError:
+        pass  # empty pupil_list, is being handled in match_data
```

Comment:
I feel like this should return an empty list in this case for consistency.

Reply:
You could even catch this case at the beginning:
```python
def filter_confidence(pupil_list, threshold):
    if not pupil_list:
        return pupil_list
    # ...
```
And get rid of all the try-catch-else stuff.